joomla/joomla-cms
# Routing: Legacy Router Rule for com_newsfeeds

This implements the current routers behavior of com_newsfeeds as a legacyrouter. This is meant for those that want to keep the old behavior at all costs. Later, parameters will be introduced to switch between the legacy behavior and new features.

This router is copied verbatim from the current router to make sure that the same behavior is kept. No changes should be done to it until Joomla 4.0, when this is supposed to be deleted. The component routers right now are so fragile, that a slow transition will be next to impossible. To prevent any breaks in backwards compatibility, this legacy router is introduced, so that people having problems with the new router can switch back.
### How to test
1. Check your site before applying the patch and see the URLs of com_newsfeeds.
2. Apply the patch.
3. See the URLs of com_newsfeeds to stay the same.
### Dependencies

This PR requires #5446 to be accepted by the project before it can be applied.

This was made possible through the generous donation of the people mentioned in the following link via an Indiegogo campaign: http://joomlager.de/crowdfunding/5-contributors
